### PR TITLE
Add an `exportable` option to default-plugins

### PR DESCRIPTION
**Pull request by kshinn**
Addresses issue #108

This simply adds and exportable parameter to the EcdsaKeyPair class to allow the key to be exported.

### CODE REVIEW

**Comment by matheus23** [comment]
> With the provided fix, I can successfully export the key, however it exports it in pkcs8 format. There doesn't seem to be a symmetrical way to import this key material. The import function uses the jwk format instead of a raw format. In order to make this change useful, we would need to also add an importFromRaw type of method to this class. This may start to diverge from the original intent of these classes.

The `export` function hard-codes the `pkcs8` format:
https://github.com/ucan-wg/ts-ucan/blob/4b1be878919de549f10654cefb3b7464409962d1/packages/default-plugins/src/p256/keypair.ts#L73-L76

Perhaps it makes sense to change it to `jwk`. It's not really a breaking change, because you weren't able to export P-256 keys previously at all, anyways.

**Comment by matheus23** [comment]
It *is* weird that it wouldn't match the RSA export format though (which is also pkcs8).

**Comment by kshinn** [comment]
To me, it looks like ed25519 is the only plugin with a full / symmetric import-export support. The p256 looks asymmetric (import is jwk, export is pkcs8). RSA only looks to export and doesn't have an import. As part of this PR, I could revamp both p256 and RSA to have an import and export based on pkcs8. What do you think?

**Comment by matheus23** [comment]
Sounds good. I'll leave it up to you to whether it's going to be JWK or PCKS8 :) :v:

**Comment by kshinn** [comment]
Ok, I did a pass and got symmetric import / export functionality for all of the plugins. I tried to reuse where I could, but realized a lot of the import type functions are there for other purposes (usually in the process of signing or verifying a message).  I went with JWK where it was available (p256 and rsa). Ed25519 is using a different library and doesn't appear to support JWK, but I it was the easiest to get working with the existing format.

File: packages/default-plugins/tests/ecdsa.test.ts
Comment (matheus23) [inline packages/default-plugins/tests/ecdsa.test.ts:88]: Export returns a `string`, but import takes a `PrivateKeyJWK`. Perhaps both should be based around `PrivateKeyJWK`? Or `string`?
I'd be happier with a change of return type in `export()` rather than trying to be backwards-compatible. What do you think?

Comment (kshinn) [reply packages/default-plugins/tests/ecdsa.test.ts:88]: Yeah, it's a bit awkward. I would prefer to normalize this on either `PrivateKeyJWK` or the more general `JsonWebKey`. The former is a bit more descriptive, the latter a bit more interoperable. Any strong feelings here?

**Comment by matheus23** [comment]
@kshinn is there a reason why you removed the `uint8arrays` dependency?
`Buffer` is only available in nodejs, and this project also targets working in browsers, so relying on `Buffer` is a no-go for its goals, unfortunately.

**Comment by kshinn** [comment]
Oops, that wasn't supposed to commit to this branch. I'll return the dependency. Long story short: I'm trying to make this library work in a constrained nodejs setting (total size needs to be as small as possible). I'm trying to eliminate some of the polyfills and work directly with node libs to do this.

But this is unrelated to the changes I'm making here and I didn't see that they leaked in. I will undo those changes in this branch.

File: packages/default-plugins/tests/ecdsa.test.ts
Comment (matheus23) [reply packages/default-plugins/tests/ecdsa.test.ts:89]: No strong feelings, but leaning towards `PrivateKeyJWK`, because the string representation is just a `JSON.stringify` away for users. That way users can opt-in to the version that's interoperable, but with worse types.

File: packages/default-plugins/src/ed25519/keypair.ts
Comment (matheus23) [inline packages/default-plugins/src/ed25519/keypair.ts:60]: Thanks for also aligning the `EdKeypair` to export as `PrivateKeyJwk`.
Do you mind adding a comment with a reference/link to something that specifies/explains why the format should be `kty`/`crv`/`d`? :pray:

**Review by matheus23 — APPROVED** [review]
Putting this on approve, because I trust you to fix the last nit before we merge :)

**Review by matheus23 — APPROVED** [review]
Thanks for your work, this looks good to me :)

**Pull request merged**
